### ISSUE TITLE
Getting 404 errors every time on cht.exe, same command works using curl

### ISSUE THREAD

**Issue by jbrown123** [tpanj/cht.exe#7]
I have the latest cht.exe (just downloaded moments ago) in my path. No matter what I request or how I request it I always get a 404 error.

When I uses `cht c/strlen` I get the following:
```
c:\>cht c/strlen
/*
 * 404 NOT FOUND
 *
 * Unknown cheat sheet. Please try to reformulate your query.
 * Query format:
 */

 /LANG/QUESTION

/* Examples: */

 /python/read+json
 /golang/run+external+program
 /js/regex+search

/*
 * See /:help for more info.
 *
 * If the problem persists, file a GitHub issue at
 * github.com/chubin/cheat.sh or ping @igor_chubin
 */
```
Asking with space instead of a slash also doesn't work.
```
c:\>cht c strlen
/*
 * 404 NOT FOUND
 *
 * Unknown cheat sheet. Please try to reformulate your query.
 * Query format:
 */

 /LANG/QUESTION

/* Examples: */

 /python/read+json
 /golang/run+external+program
 /js/regex+search

/*
 * See /:help for more info.
 *
 * If the problem persists, file a GitHub issue at
 * github.com/chubin/cheat.sh or ping @igor_chubin
 */
```

Same if I just ask for a unix command like shutdown:
```
c:\>cht shutdown
/*
 * 404 NOT FOUND
 *
 * Unknown cheat sheet. Please try to reformulate your query.
 * Query format:
 */

 /LANG/QUESTION

/* Examples: */

 /python/read+json
 /golang/run+external+program
 /js/regex+search

/*
 * See /:help for more info.
 *
 * If the problem persists, file a GitHub issue at
 * github.com/chubin/cheat.sh or ping @igor_chubin
 */
```

However, if I use curl for the same queries, it works fine:
```
c:\>curl cht.sh/shutdown
 cheat:shutdown
# To reboot the system immediately:
shutdown -r now

# To shut system down immediately:
shutdown -h now

# To reboot system after 5 minutes:
shutdown -r +5

# To cancel a scheduled shutdown:
shutdown -c

 tldr:shutdown
# shutdown
# Shutdown and reboot the system.
# More information: <https://manned.org/shutdown.8>.

# Power off (halt) immediately:
shutdown -h now

# Reboot immediately:
shutdown -r now

# Reboot in 5 minutes:
shutdown -r +5 &

# Shutdown at 1:00 pm (Uses 24h clock):
shutdown -h 13:00

# Cancel a pending shutdown/reboot operation:
shutdown -c
```

or by language

```
c:\>curl cht.sh/c/strlen
/*
 * If you want to rewrite standard C function `strlen` then you should
 * follow [the declaration
 * conventions](http:www.iso-9899.info/n1570.html#7.24.6.3) of the
 * function.
 *
 * >     size_t strlen(const char *s);
 *
 * First of all it should have return type `size_t` and the parameter
 * shall have qualifier `const` that you could use the function with
 * constant character arrays.
 * Also such a name as `sum` is not very suitable for this function. In
 * your function you have too many local variables.
 *
 * I would write the function the following way
 */

 size_t strlen( const char *s )
 {
     size_t n = 0;

     while ( s[n] ) ++n;

     return n;
 }

/* The other way is to use only pointers. For example */

 size_t strlen( const char *s )
 {
     const char *p = s;

     while ( *p ) ++p;

     return p - s;
 }

/* [Vlad from Moscow] [so/q/26923117] [cc by-sa 3.0] */
```

**Comment by jbrown123** [comment]
## Updated to add - looking in wireshark I see the following:
`cht shutdown` -> `GET /shutdown//` (note the two trailing slashes) which produces a 404 error as shown above.

On the other hand ...
`cht -Q shutdown` -> `GET /shutdown` which produces the correct results:
```
c:\>cht -Q shutdown
 cheat:shutdown
# To reboot the system immediately:
shutdown -r now

# To shut system down immediately:
shutdown -h now

# To reboot system after 5 minutes:
shutdown -r +5

# To cancel a scheduled shutdown:
shutdown -c

 tldr:shutdown
# shutdown
# Shutdown and reboot the system.
# More information: <https://manned.org/shutdown.8>.

# Power off (halt) immediately:
shutdown -h now

# Reboot immediately:
shutdown -r now

# Reboot in 5 minutes:
shutdown -r +5 &

# Shutdown at 1:00 pm (Uses 24h clock):
shutdown -h 13:00

# Cancel a pending shutdown/reboot operation:
shutdown -c
```

**Comment by cobrabr** [comment]
Same exact problem for me.

**Comment by tpanj** [comment]
Thanks all for reporting and voicing up. Project is not dead. Just little inactive due lack of time.  Need also to rethink this clumsy -Q option and correct right.  Any ideas?

Is it also true I am thinking about rewriting to vlang.

**Comment by tpanj** [comment]
@liudonghua123 AFIK flutter or tauri are more for GUI defelopment, tauri is better as electron alternative and with this we lost support for oldest computers  console (it is Win7+).  How important is this for you to have it backward compatible with XP+?

**Comment by robbely** [comment]
This is still a problem a year and a half later, so I guess this has now been abandoned :(
(Also with the -Q option - this sometimes works, sometimes crashes)

**Comment by tpanj** [comment]
Thanks for your interest and feedbacks. I am working on porting complete source code and make it more maintainable and portable vlang - I decided basically by #10 . Stay tuned since I will finish publish it in approximately month.

**Comment by tpanj** [comment]
> I have the latest cht.exe (just downloaded moments ago) in my path. No matter what I request or how I request it I always get a 404 error.
> 
> When I uses `cht c/strlen` I get the following:
> 
> ```
> c:\>cht c/strlen
> /*
>  * 404 NOT FOUND
> ```

Should work now and get the same result as `curl cht.sh/c/strlen`

> ```
> c:\>cht c strlen
> ```

This should be now applied with "domain language" option `-l` so full command for this is:
 ```
cht -l c strlen
 ```

Believe that the next versions can be more error tolerant, but for now this is it.
I would be very happy receiving feedback (so can I close issue),  test cases or more usage examples.

**Comment by robbely** [comment]
Many thanks - new version is working well for me (tested basic cht xxxx where xxx is anything I would expect in cheat.sh).

I do get an odd error "V panic: array.get: index out of range (i == 0, a.len == 0)" when i use cht with something not in the database so maybe it needs to catch this error, (but it's a trivial issue as it still does what it should).

**Comment by Gaweringo** [comment]
The scoop version is still at 0.6. Should I update the manifest to 0.7.1 and add autoupdate?

**Comment by tpanj** [comment]
> The scoop version is still at 0.6. Should I update the manifest to 0.7.1 and add autoupdate?

Great but hold on - new version is coming.
Do you think it would be better to have "_v0.7.2.zip" at end of assets for autoupdatesand so or should I leave like it is?

**Comment by Gaweringo** [comment]
> Do you think it would be better to have "_v0.7.2.zip" at end of assets for autoupdatesand so or should I leave like it is?

Whichever you like more, for the scoop auto-updater it doesn't matter, since it gets the version from the release feed.
```json
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/tpanj/cht.exe/releases/download/v$version/cht_windows.zip"
            }
        }
    }
```

> new version is coming.
Well, with auto-update set up correctly that shouldn't be a problem, as long as the file name does not change.
If you want to change the name of the zip, I'll wait.

**Comment by tpanj** [comment]
Great. Pushed new version and leave zips like they are.
